test/auth/auth.spec.js

```diff
@@ -0,0 +1,116 @@
+/* eslint-disable no-undef */
+let chai = require('chai')
+
+let chaiHttp = require('chai-http')
+
+// eslint-disable-next-line no-unused-vars
+let server = require('../../src/server.js')
+
+const { editUser } = require('../../src/controllers/user/handlers')
+
+let should = chai.should()
+
+chai.use(chaiHttp)
+
+var data = require('./shared.js')
+
+const application = require('../../src/app')
+
+const { PORT } = process.env
+let app = 'http://localhost:' + PORT
+
+let basePath = '/v1/auth'
+
+describe('Testing auth', () => {
+  describe('test auth/register-local', () => {
+    /**
+     * Test creating new user
+     */
+    describe('Test creating new user (customer) ', () => {
+      it('Creating new user (customer) ', (done) => {
+        chai
+          .request(app)
+          .post(basePath + '/register-local')
+          .send(data.customer)
+          .end((err, res) => {
+            should.not.exist(err)
+            res.should.have.status(201)
+            res.body.should.be.a('object')
+            res.body.email.should.equal(data.customer.email)
+            res.body.isCoach.should.equal(false)
+            res.body.isAdmin.should.equal(false)
+            res.body.isArchived.should.equal(false)
+            data.customer.user = res.body
+            done()
+          })
+      })
+    })
+
+    /**
+     * Test creating new Admin
+     */
+    describe('Test creating new user (admin) ', () => {
+      it('Creating new user (admin) && update isAdmin => true ', (done) => {
+        chai
+          .request(app)
+          .post(basePath + '/register-local')
+          .send(data.admin)
+          .end(async (err, res) => {
+            should.not.exist(err)
+            res.should.have.status(201)
+            res.body.should.be.a('object')
+            await editUser(res.body._id, { isAdmin: true })
```

Comment:
That's a behaviour to avoid. You are changing the logic of the API here. The goal of testing the api is only to request routes and see is everything is working as intent.